core/src/com/po/fuck/model/sprites/WeaponSpriteInfo.java

```diff
@@ -0,0 +1,23 @@
+package com.po.fuck.model.sprites;
+
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.utils.JsonValue;
+
+public class WeaponSpriteInfo extends BasicSpriteInfo {
+    public final Vector2 muzzlePosition;
+
+    public final Vector2 holdingPosition;
+/*
+    * muzzlePositionX/Y - the X/Y coordinate on the frame of the muzzle
```

Comment:
`muzzlePositionX/Y` then why are you doing `muzzlePosition.sub(new Vector2(getWidth(), getHeight()/2));` in a constructor ?

Reply:
Because of the specificity of the coordinates representing the texture position and the guns texture itself.